File: src/domain/search_query_test.py
import pytest
from src.domain.search_query import SearchQuery, QueryType


def test_ean_13_detected():
    q = SearchQuery.from_raw("8710447308431")
    assert q.type == QueryType.EAN
    assert q.raw == "8710447308431"


def test_ean_8_detected():
    q = SearchQuery.from_raw("01234565")
    assert q.type == QueryType.EAN


def test_short_number_is_text():
    q = SearchQuery.from_raw("123")
    assert q.type == QueryType.TEXT


def test_product_name_is_text():
    q = SearchQuery.from_raw("peanut butter")
    assert q.type == QueryType.TEXT


def test_mixed_string_is_text():
    q = SearchQuery.from_raw("ABC123")
    assert q.type == QueryType.TEXT


def test_empty_raises():
    with pytest.raises(ValueError, match="Query too short"):
        SearchQuery.from_raw("")


def test_single_char_raises():
    with pytest.raises(ValueError, match="Query too short"):
        SearchQuery.from_raw("a")


def test_ean_14_digits_is_text():
    q = SearchQuery.from_raw("12345678901234")
    assert q.type == QueryType.TEXT

File: frontend/src/domain/SearchQuery.test.ts
import { describe, it, expect } from "vitest";
import { createSearchQuery, QueryType } from "./SearchQuery";

describe("createSearchQuery", () => {
  it("detects EAN-13", () => {
    const q = createSearchQuery("8710447308431");
    expect(q.type).toBe(QueryType.EAN);
    expect(q.raw).toBe("8710447308431");
  });

  it("detects EAN-8", () => {
    expect(createSearchQuery("01234565").type).toBe(QueryType.EAN);
  });

  it("treats short number as text", () => {
    expect(createSearchQuery("123").type).toBe(QueryType.TEXT);
  });

  it("treats product name as text", () => {
    expect(createSearchQuery("peanut butter").type).toBe(QueryType.TEXT);
  });

  it("throws on empty string", () => {
    expect(() => createSearchQuery("")).toThrow("Query too short");
  });

  it("throws on single char", () => {
    expect(() => createSearchQuery("a")).toThrow("Query too short");
  });
});
